rust/experimental/query_engine/engine-recordset/src/scalars/text_scalar_expressions.rs

```diff
@@ -51,6 +51,164 @@ where
+            let capture_value =
+                execute_scalar_expression(execution_context, e.get_capture_name_or_index())?;
+
+            // Compile the regex pattern
+            let regex = match regex::Regex::new(pattern_string) {
+                Ok(regex) => regex,
+                Err(err) => {
+                    return Err(ExpressionError::ValidationFailure(
+                        e.get_query_location().clone(),
+                        format!("Invalid regex pattern '{}': {}", pattern_string, err),
+                    ));
+                }
+            };
+
+            // Apply the regex to the input string and extract the capture group
+            let v = if let Some(captures) = regex.captures(input_string) {
+                match capture_value.to_value() {
+                    Value::String(s) => {
+                        // Named capture group
+                        let capture_name = s.get_value();
+                        if let Some(matched) = captures.name(capture_name) {
+                            ResolvedValue::Computed(OwnedValue::String(StringValueStorage::new(
+                                matched.as_str().into(),
+                            )))
+                        } else {
+                            // Named capture group not found - return empty string
+                            ResolvedValue::Computed(OwnedValue::String(StringValueStorage::new(
+                                "".into(),
+                            )))
+                        }
+                    }
+                    Value::Integer(i) => {
+                        // Indexed capture group
+                        let index = i.get_value() as usize;
+                        if let Some(matched) = captures.get(index) {
+                            ResolvedValue::Computed(OwnedValue::String(StringValueStorage::new(
+                                matched.as_str().into(),
+                            )))
+                        } else {
+                            // Indexed capture group not found - return empty string
+                            ResolvedValue::Computed(OwnedValue::String(StringValueStorage::new(
+                                "".into(),
+                            )))
+                        }
+                    }
+                    _ => {
+                        return Err(ExpressionError::ValidationFailure(
+                            e.get_query_location().clone(),
+                            format!(
+                                "Text extract capture name or index must be a string or integer, got: '{:?}'",
+                                capture_value.get_value_type()
+                            ),
+                        ));
+                    }
+                }
+            } else {
+                // No match found - return empty string
+                ResolvedValue::Computed(OwnedValue::String(StringValueStorage::new("".into())))
```

Comment:
No way to distinguish between a match of empty capture or failure. Could return `null` for failure to make that possible. Not sure which is best.